go/coding_standard.md

```diff
@@ -31,18 +31,29 @@ Please add items here, or modify them as appropriate. We have git versioning, do
  * Many comments are not sentences, these should begin with a lower case letter and end without a period.
  * The first letter of sentences in comments are capitalized and ends with a period.
  
-## Git
-
- * Production repos never push to `master` or `develop` branches except the push-master.
- * `develop` should always pass tests.
- * Always start work with a "WIP: \*" (work-in-progress) pull-request, on a branch ideally named "feature/\*" or "hotfix/\*". You're encouraged to fork the repo and add it as a remote in .git/config (so you can push to your fork, but still use the same import paths to the original project).
- * When complete, pull requests should be opened to the `develop` branch.
- * Do not merge PRs that aren't final -- e.g. no code dead code or code that won't be used should be committed.
- * Squash and merge to keep the history clean.
- * Merging from `develop` to `master` happens during a release.
- * All branches should be in lowercase, underscores are okay in branch names
- * All pull requests should include any relevant additions to the `CHANGELOG.md`, If PR is to the develop branch changes
- can be included under an `unreleased changes` header at the top of the document.
+## Github
+
+ * Production branches
+   * Production repos never push to `master` or `develop` branches except the push-master.
+   * `develop` should always pass tests.
+   * `develop` branch should have an on going open PR to `master`, high level comments can be made here
+   * Merging from `develop` to `master` happens during a release.
+ * Feature branches
+   * should following the naming convention `yourname/branch-description`
+   * all branches should be in lowercase, underscores/dashes are okay in branch names
+   * should have an open PR with the name "WIP: your feature description"
+   * All feature PRs should opened to merge to the `develop` branch, unless stacked on other PR leading to `develop`.
+    * Ideally, for faster merges, break your PR into several PRs that stack on top of each other and lead to `develop`.
```

Comment:
I don't get this. Assuming there's a chain of PRs: `develop` <- `PR1` <- `PR2`. What happens with `PR2` when `PR1` is merged into `develop`? Is it going to be open against `develop`? If that's the case, this is an interesting way of stacking PRs.

Overall this seems like a workaround for the slowness of PR approvals. Instead of working around it, let's try to see how we can improve that and fix it.

Reply:
I think PRs should be independent and scopes should be defined so that this is possible.

Reply:
If done advocated like this, it will lead to long-running feature branches, which are prone to diverge to the point that siginificant energy needs to be spent to prepare them for merge. Instead of stacking the work should be split up in reasonable chunks that can build of one another but make for mergable changes to develop. High-level progress spanning PRs can always be tracked in a dedicated issue.

Reply:
Ok, this is one that seems to require more discussions. @rigelrozanski do you have an example where this was useful? Did it create more confusion? Did it help in chunking your work?

My opinion is what @xla said above: independent PRs, that can build on one another but make for mergeable changes to develop.

To be honest, I would even add that commit cleanup is required before merging a PR:
1. All checkpoint commits should be squashed. (Commits that are created to "save" your work but would result in an unstable/unbuildable code.)
2. All regular ("stable") commits should go into the merge.

I'm not trying to enforce this, but it's a good coding habit.

I'm open for more information.